parser-typechecker/src/Unison/Codebase/SqliteCodebase/Operations.hs

```diff
@@ -655,6 +655,48 @@ updateNameLookupIndex getDeclType pathPrefix mayFromBranchHash toBranchHash = do
         ct <- getDeclType ref
         pure (referent, Just $ Cv.constructorType1to2 ct)
 
+-- | Add an index for the provided branch hash if one doesn't already exist.
+ensureNameLookupForBranchHash ::
+  (C.Reference.Reference -> Sqlite.Transaction CT.ConstructorType) ->
+  -- | An optional branch which we may already have an index for.
+  -- This should be a branch which is relatively similar to the branch we're creating a name
+  -- lookup for, e.g. a recent ancestor of the new branch. The more similar it is, the faster
+  -- the less work we'll need to do.
+  Maybe BranchHash ->
+  BranchHash ->
+  Sqlite.Transaction ()
+ensureNameLookupForBranchHash getDeclType mayFromBranchHash toBranchHash = do
+  Ops.checkBranchHashNameLookupExists toBranchHash >>= \case
+    True -> pure ()
+    False -> do
+      (fromBranch, mayExistingLookupBH) <- case mayFromBranchHash of
+        Nothing -> pure (V2Branch.empty, Nothing)
+        Just fromBH -> do
+          Ops.checkBranchHashNameLookupExists fromBH >>= \case
+            True -> (,Just fromBH) <$> Ops.expectBranchByBranchHash fromBH
+            False -> do
+              -- TODO: We can probably infer a good starting branch by crawling through
```

Comment:
I'll probably do this soonish in the future, but for now using the previous branch head is going to be the best in most cases anyways.